### PR TITLE
AMLS-3802 Add/remove MSB sub-sectors

### PR DESCRIPTION
**Pull request by elkdanger**

### CODE REVIEW

File: app/controllers/businessmatching/MsbSubSectorsController.scala [hunk 1]
```diff
@@ -59,59 +61,14 @@ class MsbSubSectorsController @Inject()(val authConnector: AuthConnector,
...
-              Redirect(routes.PSRNumberController.get(edit))
-            } else {
-              Redirect(routes.SummaryController.get())
-            })
-
-            updateModel flatMap { _ => redirectResult } getOrElse InternalServerError("Could not update services")
+            dataCacheConnector.update[ChangeSubSectorFlowModel](ChangeSubSectorFlowModel.key) {
+              _.getOrElse(ChangeSubSectorFlowModel()).copy(subSectors = Some(data.msbServices))
+            } flatMap {
+              case Some(m@ChangeSubSectorFlowModel(Some(set), _)) if !(set contains TransmittingMoney) =>
```
Comment (JGRobinson) [inline app/controllers/businessmatching/MsbSubSectorsController.scala]: Not sure why you don't update the sector information for transmitting money

File: app/controllers/businessmatching/PSRNumberController.scala [hunk 2]
```diff
@@ -51,21 +57,23 @@ class PSRNumberController @Inject()(val authConnector: AuthConnector,
...
-          }
-          case ValidForm(_, _) =>
-            Future.successful(Redirect(routes.NoPsrController.get()))
+
+          case ValidForm(_, data) =>
+            dataCacheConnector.update[ChangeSubSectorFlowModel](ChangeSubSectorFlowModel.key) {
+              _.getOrElse(ChangeSubSectorFlowModel.empty).copy(psrNumber = Some(data))
+            } flatMap {
+              case Some(m@ChangeSubSectorFlowModel(_, Some(BusinessAppliedForPSRNumberYes(_)))) =>
+                helper.updateSubSectors(m) flatMap { _ =>
```
Comment (JGRobinson) [inline app/controllers/businessmatching/PSRNumberController.scala]: Why is this not just calling updateBusinessMatching?

File: app/controllers/msb/WhatYouNeedController.scala [hunk 3]
```diff
@@ -16,17 +16,42 @@
...
+    implicit authContext =>
+      implicit request =>
+        Future.successful(Ok(what_you_need()))
+  }
+
+  def post = Authorised.async {
+    implicit authContext =>
+      implicit request =>
+        dataCacheConnector.fetch[ServiceChangeRegister](ServiceChangeRegister.key) flatMap {
+          case Some(register) if register.addedActivities.fold(false)(_.contains(MoneyServiceBusiness)) =>
```
Comment (JGRobinson) [inline app/controllers/msb/WhatYouNeedController.scala]: Not sure why this additional check for addedActivities is here - isn't this only for MSB?

File: app/controllers/businessmatching/MsbSubSectorsController.scala (hunk 1 above)
Comment (nonimay) [reply app/controllers/businessmatching/MsbSubSectorsController.scala]: if you have transmitting money then you don't do the save until submission of the psr number page

**Pull request merged**